imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -465,35 +460,29 @@ def get_unit_vector(
 
     Returns
     -------
-    vhat_x : np.array
-        Normalized component of the velocity vector in x direction.
-    vhat_y : np.array
-        Normalized component of the velocity vector in y direction.
-    vhat_z : np.array
-        Normalized component of the velocity vector in z direction.
+    velocities : np.ndarray
+        N x 3 array of velocity components (vx, vy, vz) in km/s.
     """
     if tof[tof < 0].any():
         logger.info("Negative tof values found.")
 
-    delta_x = front_position[0] - back_position[0]
-    delta_y = front_position[1] - back_position[1]
-
-    v_x = delta_x / tof
-    v_y = delta_y / tof
-    v_z = d / tof
+    # distances in .1 mm
+    delta_x = (front_position[0] - back_position[0]) * 0.1
+    delta_y = (front_position[1] - back_position[1]) * 0.1
+    delta_z = d * 0.1
```

Comment:
This function would be considerably cleaner if you just defined a `delta_v` vector up front and use N x 3 arrays throughout.
```suggestion
    delta_v = np.empty((len(d), 3)  # Might want to specify a dtype here
    # distances in .1 mm
    delta_v[:, 0] = (front_position[0] - back_position[0]) * 0.1
    delta_v[:, 1] = (front_position[1] - back_position[1]) * 0.1
    delta_v[:, 2] = d * 0.1
```